lib/src/common/disposable_manager.dart

```diff
@@ -186,3 +257,30 @@ abstract class LeakFlagger {
 /// the object managing it is disposed, the future will complete with
 /// an instance of this exception.
 class ObjectDisposedException implements Exception {}
+
+/// Used to invoke, and remove references to, a [Disposer] before disposal of the
+/// parent object.
+abstract class SimpleDisposable {
```

Comment:
Also the doc comment is narrow than what the purpose of this class appears to be.